packages/wrangler/src/__tests__/config/findWranglerConfig.test.ts

```diff
@@ -0,0 +1,254 @@
+	});
+
+	describe("(useRedirect: true)", () => {
+		it("should return redirected config path if no user config and a deploy config is found", async () => {
+			await seed({
+				[".wrangler/deploy/config.json"]: `{"configPath": "../../dist/wrangler.json" }`,
+				[`dist/wrangler.json`]: "DUMMY",
+				["foo/holder.txt"]: "DUMMY",
+			});
+			expect(findWranglerConfig(".", { useRedirect: true })).toEqual(
+				path.resolve(`dist/wrangler.json`)
+			);
+			expect(findWranglerConfig("./foo", { useRedirect: true })).toEqual(
+				path.resolve(`dist/wrangler.json`)
+			);
+			expect(std).toMatchInlineSnapshot(`
+				Object {
+				  "debug": "",
+				  "err": "",
+				  "info": "",
+				  "out": "",
+				  "warn": "[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mUsing redirected Wrangler configuration.[0m
+
+				  Redirected config path: \\"dist/wrangler.json\\"
+				  Deploy config path: \\".wrangler/deploy/config.json\\"
+				  Original config path: \\"<no user config found>\\"
+
+
+				[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mUsing redirected Wrangler configuration.[0m
+
+				  Redirected config path: \\"dist/wrangler.json\\"
+				  Deploy config path: \\".wrangler/deploy/config.json\\"
+				  Original config path: \\"<no user config found>\\"
+
+				",
+				}
+			`);
+		});
+
+		it("should return redirected config path if matching user config and a deploy config is found", async () => {
+			await seed({
+				[`wrangler.toml`]: "DUMMY",
+				[".wrangler/deploy/config.json"]: `{"configPath": "../../dist/wrangler.json" }`,
+				[`dist/wrangler.json`]: "DUMMY",
+				["foo/holder.txt"]: "DUMMY",
+			});
+			expect(findWranglerConfig(".", { useRedirect: true })).toEqual(
+				path.resolve(`dist/wrangler.json`)
+			);
+			expect(findWranglerConfig("./foo", { useRedirect: true })).toEqual(
+				path.resolve(`dist/wrangler.json`)
+			);
+			expect(std).toMatchInlineSnapshot(`
+				Object {
+				  "debug": "",
+				  "err": "",
+				  "info": "",
+				  "out": "",
+				  "warn": "[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mUsing redirected Wrangler configuration.[0m
+
+				  Redirected config path: \\"dist/wrangler.json\\"
+				  Deploy config path: \\".wrangler/deploy/config.json\\"
+				  Original config path: \\"wrangler.toml\\"
+
+
+				[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mUsing redirected Wrangler configuration.[0m
+
+				  Redirected config path: \\"dist/wrangler.json\\"
+				  Deploy config path: \\".wrangler/deploy/config.json\\"
+				  Original config path: \\"wrangler.toml\\"
+
+				",
+				}
+			`);
+		});
+
+		it("should error if deploy config is not valid JSON", async () => {
+			await seed({
+				[".wrangler/deploy/config.json"]: `INVALID JSON`,
+			});
+
+			let error;
+			try {
+				findWranglerConfig(".", { useRedirect: true });
+			} catch (e) {
+				error = e;
+			}
+
+			expect(normalizeString(`${error}`)).toMatchInlineSnapshot(
+				`"Error: Failed to load the deploy config at .wrangler/deploy/config.json"`
```

Comment:
The error itself does contain a `cause` which is the ParseError.
And this would get logged out (sort of) in an actual Wrangler run.
But I can make that better. Thanks

Reply:
I improved this.